### PR TITLE
Switch to ESM

### PR DESCRIPTION
**Pull request by sibiraj-s**
ESM is the future

While Trying to support both ESM and CJS with rollup still causes issues when running in ESM environment, since not all the dependencies minifier uses provides correct exports for both environments.

Hence the source is converted to ESM and CJS is supported via rollup. There is no breaking changes from `v7.0.0-alpha.0` except the node requirement is slightly strict.

### CODE REVIEW

File: src/htmlparser.js [hunk 1]
```diff
@@ -431,138 +427,3 @@ class HTMLParser {
     }
   }
 }
-
-exports.HTMLParser = HTMLParser;
-exports.HTMLtoXML = function (html) {
```
Comment (DanielRuf) [inline src/htmlparser.js:436]: Are these functions unused and undocumtented? Because I'm unsure why we remove them here. Or maybe I oversaw something.

Comment (sibiraj-s) [reply src/htmlparser.js:436]: > Are these functions unused and undocumtented?

Yeah, both. They are used no-where and mentioned no-where.

Comment (sibiraj-s) [reply src/htmlparser.js:436]: May be let's not remove them yet, till something is conclusive?

Comment (sibiraj-s) [reply src/htmlparser.js:436]: Removed for now. lets deal with that later.

**Comment by sibiraj-s** [comment]
Let's ship this.

**Review by DanielRuf — APPROVED** [review]

**Pull request merged**